### ISSUE TITLE
Devise::Models::Lockable::ClassMethods#unlock_strategy_enabled? always returns true for :both strategy

### ISSUE THREAD

**Issue by f3ndot** [plataformatec/devise#4072]
### Tested on:
- Ruby 2.2.2
- Rails 4.2.4
- Devise 3.5.8
- Warden 1.2.6

### Problem:
[`Devise::Models::Lockable::ClassMethods#unlock_strategy_enabled?`](http://www.rubydoc.info/github/plataformatec/devise/master/Devise/Models/Lockable/ClassMethods#unlock_strategy_enabled%3F-instance_method) always returns `true` for `:both` strategy

Note `:time`, `:email`, and `:none` do not exhibit this bug.

This can be problem for those who choose to extend the functionality of Lockable or rely on the method to determine code paths based on the `:none` strategy.

## Steps to Reproduce

1. Create a minimal Rails 4 app
2. Add Devise gem, run the devise install generator, generate a devise user
3. Add the `:lockable` module for user, uncomment the lockable migration lines and run migrations
4. Go into devise initializer and ensure `config.unlock_strategy = :both` is set
5. Boot up Rails console

## Expected

```
2.2.2 :001 > User.unlock_strategy_enabled? :none
 => false
2.2.2 :001 > User.unlock_strategy_enabled? :sadsa
 => false
2.2.2 :002 > User.unlock_strategy_enabled? :both
 => true
2.2.2 :003 > User.unlock_strategy_enabled? :time
 => true
2.2.2 :004 > User.unlock_strategy_enabled? :email
 => true
```

## Actual

```
2.2.2 :001 > User.unlock_strategy_enabled? :none
 => true
2.2.2 :001 > User.unlock_strategy_enabled? :sadsa
 => true
2.2.2 :002 > User.unlock_strategy_enabled? :both
 => true
2.2.2 :003 > User.unlock_strategy_enabled? :time
 => true
2.2.2 :004 > User.unlock_strategy_enabled? :email
 => true
```

As you can see `:sadsa` and `:none` were considered `true`.

**Comment by rishi-freshbooks** [comment]
👍

**Comment by ulissesalmeida** [comment]
Fixed on `4.1.0`, `4.0.2` and `3.5.9`.
